EIPS/eip-2315.md

```diff
@@ -13,55 +13,67 @@ requires: 3540, 3670, 4200
 > 
-> — Alan Turing — *cf. B.E. Carpenter, R.W. Doran, "The other Turing machine." The Computer Journal, Volume 20, Issue 3, January 1977.*
+> — Alan Turing — in [B.E. Carpenter, R.W. Doran, "The other Turing machine." The Computer Journal, Volume 20, Issue 3, DOI: 10.1093/comjnl/20.3.269, January 1977.](https://academic.oup.com/comjnl/article/20/3/269/751954)
+
+In more contemporary terms, we have sequences of instructions, jumps and conditional jumps that divide sequences into blocks, subroutine calls, and a stack of addresses to return to.  The details vary, but similar facilities have proven their value across a long line of important machines over the last 75 years, from the ACE to the Cray to the ARM.
 
-In more contemporary terms, we have sequences of instructions, jumps and conditional jumps that divide sequences into blocks, subroutine calls, and a stack of addresses to return to.  The details vary, but similar facilities have proven their value across a long line of machines going back over 75 years.
+Unlike these machines, the Ethereum Virtual Machine _does not_ provide subroutine operations.  Instead, they must be synthesized using the dynamic `JUMP` instruction, which takes its destination on the stack. Further, the EVM provides _only_ dynamic jumps, impeding the static analysis we need.
 
-Unlike these machines, the Ethereum Virtual Machine _does not_ provide subroutine operations.  Instead, they must be synthesized using the dynamic `JUMP` instruction, which takes its destination on the stack. Further, the EVM provides _only_ dynamic jumps, preventing the static analysis we need.
+### We need safe control-flow.
 
-### We want a safer control-flow facilty.
+The EVM has unusually high requirements for safety.  Not only do many smart contracts control inordinately large amounts of valuable Ether, but once placed on the blockchain any defects are visible to attackers and cannot be repaired.  In valid code is an  We need to statically validate important safety constraints code at initialization time.
 
-The EVM has unusually high requirements for safety.  Not only do many smart contracts control inordinately large amounts of valuable Ether, but once placed on the blockchain any defects are visible to attackers and cannot be repaired.  We propose here to statically validate, at initialization time, that code will not execute invalid instructions, jump to invalid locations, or underflow stack.  It will maintain consistent numbers of inputs and outputs for subroutines, and stack growth is bounded in the absence of recursion.
+### We need static control-flow.
```

Comment:
```suggestion
### The control-flow needs to be static
```
```suggestion
###  The control-flow needs to be safer
```

Optional, just to see if this way puts more emphasize on the characteristic this EIP intend to achieve

Reply:
Versus keeping section titles short and avoiding passive voice.